src/emitter.ts

```diff
@@ -1207,17 +1210,57 @@ export function emitWebIdl(webidl: Browser.WebIdl, flavor: Flavor) {
             return `extends ${result} `;
         }
 
+        function hasSequenceArgument(s: Browser.Signature) {
+            function typeIncludesSequence(type: string | Browser.Typed[]): boolean {
+                if (Array.isArray(type)) {
+                    return type.some(t => typeIncludesSequence(t.type))
+                }
+                return type === "sequence" || !!sequenceTypedefMap[type];
+            }
+            return !!s.param && s.param.some(p => typeIncludesSequence(p.type));
+        }
+
+        function filterSignaturesWithSequence(signature: Browser.Signature[]): Browser.Signature[] {
+            return signature.filter(hasSequenceArgument).map(s => {
+                const newParams = s.param!.map(p => {
+                    const typedef = typeof p.type === "string" ? sequenceTypedefMap[p.type]: undefined;
```

Comment:
you could drop lines 1227-1229 if you use `p.type` instead of `undefined`. It's less efficient but that's not a concern here.

```suggestion
                    const typedef = typeof p.type === "string" ? sequenceTypedefMap[p.type] : p.type;
```

Reply:
Doing this will break code as `sequenceTypedefMap[p.type]` also can be `undefined`.

Reply:
That makes sense.